### DISCUSSION TITLE
question on recommendation

### DISCUSSION THREAD

**Discussion by kimpenhaus** [Koenkk/zigbee2mqtt#15493] [General]
hi all,

I am fairly new to zigbee2mqtt and I am not pretty sure what the best way is to achieve what I like to. So any help is highly appreciated :)

My setup is as follows:

I am running zigbee2mqtt `1.28.2` on a pi - connected is a `SONOFF Zigbee 3.0 USB Dongle Plus ZigBee 3.x.0 TI CC2652P + CP2102N Coordinator` and a mqtt bus running on the local network.

Joined device is an Aqara motion detector - in the light switch there is a shelly dimmer 2.

What I currently have is, that when motion is detected the light is turned on (via home assistant). After 60 seconds this occupancy is cleared by zigbee2mqtt and via home assistant the light is turned off again.

<img width="1020" alt="image" src="https://user-images.githubusercontent.com/575109/206908788-3a06bceb-6255-438d-a67b-0272fe198c6d.png">

This okay so far - but once you are longer than 60 seconds in that room - the light goes off and on again. I'd like to have it more like once the motion is detected the event ist triggered and a timer starts.... once this timer reached its end and there is no motion in between the light goes off. But if there is any motion the timer is rested and starts again. Hopefully it is clear what my intention is (sorry - no native speaker).

Is that doable within zigbee2mqtt or should this be inside home assistant? And if so - what is the configuration I have to use? I tried the `no_occupancy_since` setting but the effect was not what I expected. Hm - just found this link: https://community.home-assistant.io/t/zigbee2mqtt-no-occupancy-since-attribute-is-not-showing-up-for-entities/426707 maybe that is/was the reason it is/was not working as expected 🤔 

Thanks for you help 👋🏼

**Comment by lux73** [comment]
you need a Timer Rule in HomeAssistant.

I'm using OpenHAB and after no occupancy occurred a timer start running - if there is occupancy while the timer is running i clearing the running timer and start a new one... and so on

with this your Light will stay on while there is occupancy

be aware a occupancy Sensor is no presence Sensor - so afaik all occupancy Sensors need to detect movement. if there's no occupancy it could be that your Sensor is resetting to no occupancy ^^

**Reply by kimpenhaus** [reply]
@lux73 thanks Michael!

So just to be clear: you turn on the light on occupancy and start the timer when there is no more occupancy? After what time is the no occupancy event fired?

Or do you handle lights by presence detection?

**Reply by lux73** [reply]
correct, Timer start when there ist no occupancy.

the time depends - If a window ist open/closed, it is evening or daytime but the brightness ist too low and so on.

i would recommend start your Timer with 1 or 2 Minute

Dont forget: If the Timer is running and new occupancy is registered you must kill the running Timer. Otherwise your will go Off

There could be a simpler solution with HA but i dont know. This one described works like a charm on OpenHAB for me 😉

**Reply by kimpenhaus** [reply]
that sounds doable :) I am not o home assistant pro but from what I understood it's easy to reset/restart the timer by just calling its start method again: https://www.home-assistant.io/integrations/timer/

thanks for all your help Michael!

**Reply by lux73** [reply]
You're Welcome 🙂
